mm2src/coins/lp_coins.rs

```diff
@@ -2707,23 +2739,103 @@ impl Default for PrivKeyActivationPolicy {
     fn default() -> Self { PrivKeyActivationPolicy::ContextPrivKey }
 }
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub enum PrivKeyPolicy<T> {
-    KeyPair(T),
+    Iguana(T),
+    HDWallet {
+        /// Derivation path of the coin.
+        /// This derivation path consists of `purpose` and `coin_type` only
+        /// where the full `BIP44` address has the following structure:
+        /// `m/purpose'/coin_type'`.
+        derivation_path: StandardHDPathToCoin,
+        activated_key: T,
+        bip39_secp_priv_key: ExtendedPrivateKey<secp256k1::SecretKey>,
```

Comment:
Let me work on this and consider it's pros on cons in next PR since it will require some refactors. Added to checklist here https://github.com/KomodoPlatform/komodo-defi-framework/issues/1838#issuecomment-1675467570